### ISSUE TITLE
Spot all SCP Errors

### ISSUE THREAD

**Issue by mossblaser** [project-rig/rig#141]
At present, only a small number of RC errors are spotted by the SCP implementation (and the even-faster-scp-rw2 branch in #117 too). The full list of error-types in in sark.h is:

```C
#define RC_LEN 			0x81	//!< Bad packet length
#define RC_SUM 			0x82	//!< Bad checksum
#define RC_CMD 			0x83	//!< Bad/invalid command
#define RC_ARG     		0x84	//!< Invalid arguments
#define RC_PORT	 		0x85	//!< Bad port number
#define RC_TIMEOUT 		0x86	//!< Timeout
#define RC_ROUTE 		0x87	//!< No P2P route
#define RC_CPU	 		0x88	//!< Bad CPU number
#define RC_DEAD	 		0x89	//!< SHM dest dead
#define RC_BUF			0x8a	//!< No free SHM buffers
#define RC_P2P_NOREPLY	0x8b	//!< No reply to open
#define RC_P2P_REJECT	0x8c	//!< Open rejected
#define RC_P2P_BUSY		0x8d	//!< Dest busy
#define RC_P2P_TIMEOUT	0x8e	//!< Dest died?
#define RC_PKT_TX		0x8f	//!< Pkt Tx failed
```

This caught me out while playing with #117 since the result is an error packet being unpacked as if it was data...

**Comment by mundya** [comment]
Supported:

 - [ ] 0x81
 - [ ] 0x8
 - [ ] 0x8

**Comment by mundya** [comment]
This is fixed in #117.
